### PR TITLE
cmd: flag parsing cleanup

### LINKED ISSUE #83: krew -h prints only glog help
```
$ kubectl krew -h
Usage of /Users/ahmetb/.krew/bin/kubectl-krew:
  -alsologtostderr
    	log to standard error as well as files
  -log_backtrace_at value
    	when logging hits line file:N, emit a stack trace
  -log_dir string
    	If non-empty, write log files in this directory
  -logtostderr
    	log to standard error instead of files
  -stderrthreshold value
    	logs at or above this threshold go to stderr
  -v value
    	log level for V logs
  -vmodule value
    	comma-separated list of pattern=N settings for file-filtered logging
```

### PR DESCRIPTION
**Pull request by ahmetb**
- simplify flag parsing logic (fixes #83)
- hide all glog files but -v (regardless of running as plugin or not)
- call GetExecutedVersion only in "version" cmd
  - TODO: maybe remove GetExecutedVersion, not used in any decisions
- unvendor viper (and many of its dependencies) since unused.

### CODE REVIEW

**Comment by lbb** [comment]
needs rebase

**Comment by ahmetb** [comment]
rebased

File: cmd/krew/cmd/version.go [hunk 1]
```diff
@@ -35,9 +38,19 @@ IndexURI is the URI where the index is updated from.
 InstallPath is the base path for all plugin installations.
 DownloadPath is the path used to store download binaries.`,
 	RunE: func(cmd *cobra.Command, args []string) error {
+		selfPath, err := os.Executable()
+		if err != nil {
+			glog.Fatalf("failed to get the own executable path")
+		}
+
+		executedVersion, runningAsPlugin, err := environment.GetExecutedVersion(paths.InstallPath(), selfPath, environment.Realpath)
```
[the remark this answers is not in the document]
Comment (ahmetb) [reply cmd/krew/cmd/version.go]: I will take a look. Probably can be done without a global.

[the remark this answers is not in the document]
Comment (ahmetb) [reply cmd/krew/cmd/version.go]: Take a look now. I copied this call into upgrade.go and tested upgrading krew, it works.

File: cmd/krew/cmd/upgrade.go [hunk 2]
```diff
@@ -53,14 +54,18 @@ kubectl plugin upgrade foo bar"`,
 			pluginNames = args
 		}
 
+		executedKrewVersion, _, err := environment.GetExecutedVersion(paths.InstallPath(), os.Args[0], environment.Realpath)
```
Comment (lbb) [inline cmd/krew/cmd/upgrade.go]: args[0] is not the absolute binary location. We need `os.Executeable()`

**Comment by lbb** [comment]
LGTM when fixed

File: cmd/krew/cmd/upgrade.go (hunk 2 above)
Comment (ahmetb) [reply cmd/krew/cmd/upgrade.go]: > args[0] is not the absolute binary

This worked fine when I manually tested. But I'll fix.

**Pull request merged**